File: server/src/utils/pathUtil.ts
// server/src/utils/pathUtil.ts

import { fileURLToPath } from 'url';
import { dirname } from 'path';

// Convert import.meta.url to a file path
export const __filename = fileURLToPath(import.meta.url);

// Get the directory name of the current module
export const __dirname = dirname(__filename);

File: backend/core/utils/path_util.py
# core/utils/path_util.py

from pathlib import Path

# Get the absolute path of the current file
__filename = Path(__file__).resolve()

# Get the directory name of the current file
__dirname = __filename.parent

# Optional: Define functions for dynamic usage
def get_filename():
    return __filename

def get_dirname():
    return __dirname
